github/codeowners.py

```diff
@@ -116,38 +118,146 @@ def resolve_team_members(
     if not team_or_none:
         logger.warning('failed to lookup team {t}'.format(t=team_name))
         return []
-    for member in map(github_api.user, team_or_none.members()):
-        if member.email:
-            yield member.email
+    yield from map(github_api.user, team_or_none.members())
+
+
+@dataclasses.dataclass
+class CodeownerEntry:
+    value: str
+
+
+@dataclasses.dataclass
+class UsernameCodeownerEntry(CodeownerEntry):
```

Comment:
how about just.. username?

consider implementing __str__ (or just inheriting from str, even..- not much value-add from dataclasses in this case..)